bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleTriggerManager.java

```diff
@@ -217,8 +217,12 @@ public RuleTriggerManager(Injector injector) {
                     for (EventTrigger t : rule.getEventtrigger()) {
                         if (t instanceof EventEmittedTrigger) {
                             EventEmittedTrigger et = (EventEmittedTrigger) t;
-                            if (et.getTrigger() != null) {
-                                if (et.getChannel().equals(channel) && et.getTrigger().equals(event)) {
+
+                            if (et.getChannel().equals(channel)) {
+                                if (et.getTrigger() == null) {
+                                    // if the rule does not have a specific event , execute it on any event
+                                    result.add(rule);
+                                } else if ((et.getTrigger() != null && et.getTrigger().equals(event))) {
```

Comment:
The first part is redundant: you already _know_ that `et.getTrigger()` is not `null`.

Anyway, why so complicated? Wouldn't a 

```java 
if (et.getChannel().equals(channel) && (et.getTrigger() == null || et.getTrigger().equals(event))) {
```
have done the job nicely without duplicating the (admittedly trivial) inner block?